From 0f340e41ccb15f7492dbd924ad8828ef75e6dc1a Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:14:06 -0400
Subject: [powerpc] of: compiles fixes

Message-id: <1410545655-205645-618-git-send-email-dzickus@redhat.com>
Patchwork-id: 94476
O-Subject: [RHEL7 PATCH 617/626] powerpc, of: compiles fixes
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

rhel-7.0 sysfs stuff behaves a little differently
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/of/base.c b/drivers/of/base.c
index 0cc6f1d..f1fa838 100644
--- a/drivers/of/base.c
+++ b/drivers/of/base.c
@@ -177,8 +177,8 @@ static const char *safe_name(struct kobject *kobj, const char *orig_name)
  int i = 0;
 
  /* don't be a hero. After 16 tries give up */
- while (i < 16 && (kn = sysfs_get_dirent(kobj->sd, name))) {
-  sysfs_put(kn);
+ while (i < 16 && (kn = sysfs_get_dirent(kobj->sd, NULL, name))) {
+  sysfs_put(kobj->sd);
   if (name != orig_name)
    kfree(name);
   name = kasprintf(GFP_KERNEL, "%s#%i", orig_name, ++i);
-- 
1.7.1